shared/actions/authActions.js

```diff
@@ -109,50 +59,25 @@ export const grantAnonymousUser = async (secret, userKey) => {
       }
     })
     .catch((error) => {
+      if (error.message === 'wrong secret') {
+        signupGuestUser();
+      }
       console.log(error);
     });
-  } catch (error) {
-    console.log(error);
-  }
 };
 
+// @TODO need to change this function's name.
 export const grantFacebookUser = async (facebookToken) => {
-  try {
-    const address = `${HTTPS}${SERVER_ADDR}${ENDPOINT_GRANT}`;
-    const headers = DEFAULT_HEADERS;
-    const body = JSON.stringify({
-      'grantType': 'facebook',
-      'facebookToken': facebookToken
-    });
-    await fetch(address, {
-      method: 'POST',
-      headers,
-      body
-    })
-    .then((response) => {
-      if (response.status === 200) {
-        return response.json();
-      } else {
-        return null;
-      }
-    })
+  return AuthRESTManager.grantFacebook(facebookToken)
     .then((rjson) => {
-      if (rjson) {
-        setAccessToken(rjson.accessToken);
-        setRefreshToken(rjson.refreshToken);
-        setUserKey(rjson.userKey);
-      } else {
-        return signupFacebookUser(facebookToken);
-      }
+      setAccessToken(rjson.accessToken);
+      setRefreshToken(rjson.refreshToken);
+      setUserKey(rjson.userKey);
```

Comment:
what if the condition of (rjson) is false but it does not throw any error and pass it to catch? then signupFacebookUser would be not called